src/plugins/intel_gna/memory/gna_mem_requests.hpp

```diff
@@ -70,8 +72,34 @@ inline const char* rTypeToStr(uint8_t type) {
 struct MemRequest {
     rRegion  _region;
     uint8_t   _type;
+
     void *_ptr_out;
     const void *_ptr_in = nullptr;
+
+    bool IsPtrOutSet() const {
+        return _ptr_out != nullptr;
+    }
+    bool IsPtrInSet() const {
+        return _ptr_in != nullptr;
+    }
+    uint8_t* GetPtrOutValue() const {
+        IE_ASSERT(_ptr_out != nullptr);
+        return reinterpret_cast<uint8_t*>(*reinterpret_cast<void**>(_ptr_out));
+    }
+    void SetPtrOutValue(void * valueIn) {
+        IE_ASSERT(_ptr_out != nullptr);
+        *reinterpret_cast<void**>(_ptr_out) = valueIn;
+    }
+    bool IsInPtrEqualToOutPtrOf(const MemRequest& reference) const {
+        return _ptr_in == reference._ptr_out;
+    }
+    const void * GetInPtr() const {
```

Comment:
What is the purpose to throw an exception in the getter function? 
And imho we don't need getters for those fields.

Reply:
Will remove these helpers.
You are right they are not needed.
I added them to ease debugging at some point but not needed anymore.